### ISSUE TITLE
Possibility to use a custom navigation page

### ISSUE THREAD

**Issue by ShiroYacha** [UXDivers/Gorilla-Player-Support#242]
Hi,

I see that currently Gorilla only supports rendering of pages inside a navigation page configuring with the Gorilla.json file. I wonder if there could be support to register custom navigation pages (along with different renderers)?

Thanks

**Comment by LeoHere** [comment]
@ShiroYacha can you describe a bit more what are your needs? What do you mean with "along with different renderers"?

**Comment by ShiroYacha** [comment]
@LeoHere for instance, i'm using the iconize library https://github.com/jsmarcus/Iconize they have `IconToolbarItem` that use their font icon, but it only renders on `IconNavigationPage`. So with Gorilla, I can't really visualize the icons.

**Comment by LeoHere** [comment]
@ShiroYacha right, got it. You can't right now. We will consider for future release.

**Comment by LeoHere** [comment]
@ShiroYacha yesterday we released **Gorilla 1.1.0.4**. On of the features we included is what you requested here, is the ability to customize how the navigation page should be instantiated.

Essentially now the `Config` object have a property called `CustomPageWrapper` that if defined will be invoked to wrap those pages that needs to be displayed within a navigation page, for example:

~~~
var config = new Config(...);
config.CustomPageWrapper = (p) => new NavigationPage(p);
~~~ 

The code above defines the default behavior of Gorilla.

Hope it helps.

**Comment by ShiroYacha** [comment]
Hi thanks for the info. I just downloaded the new version. I didn't know cuz when I use the Gorillay player menu "Check for update" it says that it is up-to-date. Can you confirm this?

**Comment by LeoHere** [comment]
@ShiroYacha that is because we still didn't update the server to push the new version to all our users. Mainly because we are still working on the documentation for the localization feature. But you should be able to download it from the [download page](http://gorillaplayer.com/download.html).
